{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/storageG1/lwang/Projects/tb_dr_MIC_v2/nn_model_script_emb_test.py - starting\n"
     ]
    },
    {
     "ename": "RuntimeError",
     "evalue": "module was compiled against NumPy C-API version 0x10 (NumPy 1.23) but the running NumPy has C-API version 0xf. Check the section C-API incompatibility at the Troubleshooting ImportError section at https://numpy.org/devdocs/user/troubleshooting-importerror.html#c-api-incompatibility for indications on how to solve this problem.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "\u001b[0;31mRuntimeError\u001b[0m: module was compiled against NumPy C-API version 0x10 (NumPy 1.23) but the running NumPy has C-API version 0xf. Check the section C-API incompatibility at the Troubleshooting ImportError section at https://numpy.org/devdocs/user/troubleshooting-importerror.html#c-api-incompatibility for indications on how to solve this problem."
     ]
    },
    {
     "data": {
      "text/plain": [
       "<torch._C.Generator at 0x7f6bdb87cdb0>"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#%%\n",
    "print('/mnt/storageG1/lwang/Projects/tb_dr_MIC_v2/nn_model_script_emb_test.py - starting')\n",
    "\n",
    "from array import array\n",
    "from cmath import nan\n",
    "from pyexpat import model\n",
    "import statistics\n",
    "from tkinter.ttk import Separator\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import os\n",
    "import torch\n",
    "import torch.nn as nn\n",
    "import torch.nn.functional as F\n",
    "import torch.optim as optim\n",
    "# from torchviz import make_dot\n",
    "from torch.utils.data import Dataset, TensorDataset, DataLoader\n",
    "from torch.utils.data.dataset import random_split\n",
    "from torchvision import datasets, transforms\n",
    "from torch.autograd import variable\n",
    "from itertools import chain\n",
    "from sklearn import metrics as met\n",
    "import pickle\n",
    "from icecream import ic\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import pathlib\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "from sklearn.preprocessing import OneHotEncoder\n",
    "from importlib import reload\n",
    "# import util\n",
    "# import model_torch_simple\n",
    "# from torchmetrics import Accuracy\n",
    "from tqdm import tqdm\n",
    "import argparse\n",
    "from icecream import ic\n",
    "import numpy as np\n",
    "from PIL import Image\n",
    "device = 'cuda' if torch.cuda.is_available() else 'cpu'\n",
    "from sklearn.metrics import roc_auc_score, confusion_matrix, accuracy_score\n",
    "#%%\n",
    "seed = 42\n",
    "# torch.backends.cudnn.deterministic = True\n",
    "# torch.backends.cudnn.benchmark = False\n",
    "torch.manual_seed(seed)\n",
    "# torch.cuda.manual_seed_all(seed)\n",
    "\n",
    "# train_data = np.loadtxt('/mnt/storageG1/lwang/Projects/tb_dr_MIC_v2/data_new_emb/aa_data_train_gene.csv', delimiter = ',')\n",
    "# train_target = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC_v2/data_new_emb/mic_aa_train_hml.csv')\n",
    "# train_target = train_target[['EMB_MIC']]\n",
    "# # don't touch test data, split out validation data from training data during training\n",
    "# # test_data = np.loadtxt('/mnt/storageG1/lwang/Projects/tb_dr_MIC_v2/data_new_EMB/aa_data_test_pca4k.csv', delimiter = ',')\n",
    "# test_data = np.loadtxt('/mnt/storageG1/lwang/Projects/tb_dr_MIC_v2/data_new_emb/aa_data_test_gene.csv', delimiter = ',')\n",
    "# test_target = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC_v2/data_new_emb/mic_aa_test_hml.csv')\n",
    "# test_target = test_target[['EMB_MIC']]\n",
    "\n",
    "# all_data = np.concatenate((train_data, test_data), axis=0)\n",
    "# all_target = pd.concat((train_target, test_target), axis=0)\n",
    "\n",
    "# train_data, test_data, train_target, test_target = train_test_split(all_data, all_target, test_size=0.2, random_state=42, stratify=all_target)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def data_split(aa_array, encoded_mic):\n",
    "    # Encode the target variable\n",
    "    # Perform stratified train-test split\n",
    "    train_data, test_data, train_target, test_target = train_test_split(\n",
    "        aa_array,\n",
    "        encoded_mic,\n",
    "        test_size=0.1,  # 10% for testing\n",
    "        stratify=encoded_mic,  # Ensures the proportion of each class is preserved\n",
    "        random_state=42  # For reproducibility\n",
    "    )\n",
    "    return train_data, test_data, train_target, test_target\n",
    "def is_within_doubling_dilution(pred, target, target_min, target_max):\n",
    "    _ = np.arange(target_min-1, target_max+2, 1)\n",
    "    index = [i for i, x in enumerate(_) if x == target][0]\n",
    "    return (_[index-1] <= pred <= _[index+1])\n",
    "\n",
    "def transform_array(arr):\n",
    "    transformed = []\n",
    "    for item in arr:\n",
    "        # Only process string items that contain a comma\n",
    "        if isinstance(item, str) and ',' in item:\n",
    "            parts = [part.strip() for part in item.split(',')]\n",
    "            transformed.extend(parts)\n",
    "        else:\n",
    "            transformed.append(item)\n",
    "    return transformed\n",
    "\n",
    "def data_prep_cryptic(cryptic, gene_list, dr_list):\n",
    "    # overlap = set(variants['sample_id']).intersection(set(cryptic['ENA_RUN'].to_list()))\n",
    "    # variants = variants[variants['drugs'].isin(['ethambutol'])]\n",
    "    # variants = variants[variants['sample_id'].isin(overlap)]\n",
    "    # variants['SNP'] = variants['gene'] + '-'+ variants['change']\n",
    "    variants = pd.read_csv('../variants_full.csv')\n",
    "    variants = pd.read_csv('../variants_full.csv')\n",
    "    variants = variants[variants['gene'] != 'PPE35']\n",
    "    variants = variants[variants['type'] != 'synonymous_variant']\n",
    "    variants = variants[variants['type'] != 'non_coding_transcript_exon_variant']\n",
    "\n",
    "    overlap = set(variants['sample_id']).intersection(set(cryptic['ENA_RUN'].to_list()))\n",
    "    # variants = variants[variants['drugs'].isin(['ethambutol'])]\n",
    "    variants = variants[variants['gene'].isin(gene_list)]\n",
    "    variants = variants[variants['sample_id'].isin(overlap)]\n",
    "    variants['SNP'] = variants['gene'] + '-'+ variants['change']\n",
    "    # crypticSNPnames = np.load('crypticSNPnames.npy', allow_pickle=True)\n",
    "    # variants = variants[variants['SNP'].isin(crypticSNPnames)]\n",
    "    print(variants['SNP'].unique().shape)\n",
    "    # print(variants.shape)\n",
    "    # print(variants['sample_id'].unique().shape)\n",
    "    def compare_snp_lists_with_values_optimized(set_list, query_list, values_list):\n",
    "        # Create a dictionary from query_list and values_list for direct mapping\n",
    "        query_dict = dict(zip(query_list, values_list))\n",
    "        # Use list comprehension to build the output list directly\n",
    "        output_list = [query_dict.get(snp, 0) for snp in set_list]\n",
    "        return output_list\n",
    "    # Example usage\n",
    "    # set_list = ['SNP1', 'SNP2', 'SNP3', 'SNP4']\n",
    "    # query_list = ['SNP2', 'SNP4']\n",
    "    # values_list = [5, 10]  # Corresponding values for 'SNP2' and 'SNP4'\n",
    "    # output_list = compare_snp_lists_with_values_optimized(set_list, query_list, values_list)\n",
    "    # print(output_list)  # Expected output: [0, 5, 0, 10]# Getting all snp data\n",
    "    aa = []\n",
    "    dr = []\n",
    "    all_snp = variants['SNP'].unique() # here is a list of all snps values title for the row in the final table\n",
    "    np.save('all_snp_emb.npy', all_snp)\n",
    "    print(len(all_snp))\n",
    "    print(len(overlap))\n",
    "    for x in tqdm(overlap):\n",
    "    # for x in tqdm(variants['sample_id'].unique()):\n",
    "        if x in variants['sample_id'].tolist():\n",
    "            aa.append(compare_snp_lists_with_values_optimized(all_snp, variants[variants['sample_id']==x]['SNP'].to_list(), variants[variants['sample_id']==x]['freq'].to_list()))\n",
    "        else:\n",
    "            aa.append([0]*len(all_snp))\n",
    "        \n",
    "        # print('SNP')\n",
    "    aa_array = np.array(aa)\n",
    "    aa_array[aa_array < 0.8] = 0\n",
    "    aa_array[aa_array >= 0.8] = 1\n",
    "    \n",
    "    mic_aa = cryptic[cryptic['ENA_RUN'].isin(overlap)]#.iloc[:,14:27]\n",
    "    # mic_aa = cryptic[cryptic['ENA_RUN'].isin(variants['sample_id'].unique())]#.iloc[:,14:27]\n",
    "    # print(mic_aa.shape)\n",
    "    # mic_aa['wgs_id'] = pd.Categorical(mic_aa['ENA_RUN'], categories=variants['sample_id'].unique().tolist(), ordered=True)\n",
    "    # mic_aa = mic_aa.sort_values('ENA_RUN')\n",
    "    mic_aa.ENA_RUN = mic_aa.ENA_RUN.astype('category')\n",
    "    mic_aa.ENA_RUN = mic_aa.ENA_RUN.cat.set_categories(overlap)\n",
    "    # mic_aa.ENA_RUN = mic_aa.ENA_RUN.cat.set_categories(variants['sample_id'].unique().tolist())\n",
    "    mic_aa = mic_aa.sort_values([\"ENA_RUN\"])  ## 'sort' changed to 'sort_values'\n",
    "    \n",
    "    return aa_array, mic_aa\n",
    "\n",
    "def data_prep_50k(cryptic, gene_list, dr_list):\n",
    "    # overlap = set(variants['sample_id']).intersection(set(cryptic['ENA_RUN'].to_list()))\n",
    "    # variants = variants[variants['drugs'].isin(['ethambutol'])]\n",
    "    # variants = variants[variants['sample_id'].isin(overlap)]\n",
    "    # variants['SNP'] = variants['gene'] + '-'+ variants['change']\n",
    "    variants = pd.read_csv('../variants_full.csv')\n",
    "    variants = pd.read_csv('../variants_full.csv')\n",
    "    variants = variants[variants['gene'] != 'PPE35']\n",
    "    variants = variants[variants['type'] != 'synonymous_variant']\n",
    "    variants = variants[variants['type'] != 'non_coding_transcript_exon_variant']\n",
    "\n",
    "    overlap = set(variants['sample_id']).intersection(set(cryptic['ENA_RUN'].to_list()))\n",
    "    # variants = variants[variants['drugs'].isin(['ethambutol'])]\n",
    "    variants = variants[variants['gene'].isin(gene_list)]\n",
    "    variants = variants[variants['sample_id'].isin(overlap)]\n",
    "    variants['SNP'] = variants['gene'] + '-'+ variants['change']\n",
    "    # crypticSNPnames = np.load('crypticSNPnames.npy', allow_pickle=True)\n",
    "    # variants = variants[variants['SNP'].isin(crypticSNPnames)]\n",
    "    print(variants['SNP'].unique().shape)\n",
    "    # print(variants.shape)\n",
    "    # print(variants['sample_id'].unique().shape)\n",
    "    def compare_snp_lists_with_values_optimized(set_list, query_list, values_list):\n",
    "        # Create a dictionary from query_list and values_list for direct mapping\n",
    "        query_dict = dict(zip(query_list, values_list))\n",
    "        # Use list comprehension to build the output list directly\n",
    "        output_list = [query_dict.get(snp, 0) for snp in set_list]\n",
    "        return output_list\n",
    "    # Example usage\n",
    "    # set_list = ['SNP1', 'SNP2', 'SNP3', 'SNP4']\n",
    "    # query_list = ['SNP2', 'SNP4']\n",
    "    # values_list = [5, 10]  # Corresponding values for 'SNP2' and 'SNP4'\n",
    "    # output_list = compare_snp_lists_with_values_optimized(set_list, query_list, values_list)\n",
    "    # print(output_list)  # Expected output: [0, 5, 0, 10]# Getting all snp data\n",
    "    aa = []\n",
    "    dr = []\n",
    "    \n",
    "    all_snp = variants['SNP'].unique() # here is a list of all snps values title for the row in the final table\n",
    "    # np.save('all_snp_emb.npy', all_snp)\n",
    "    print(len(all_snp))\n",
    "### 50k data reload\n",
    "    variants = pd.read_csv('../variants_full.csv')\n",
    "    variants = variants[variants['gene'] != 'PPE35']\n",
    "    variants = variants[variants['type'] != 'synonymous_variant']\n",
    "    variants = variants[variants['type'] != 'non_coding_transcript_exon_variant']\n",
    "    variants['SNP'] = variants['gene'] + '-'+ variants['change']\n",
    "    variants = variants[variants['SNP'].isin(all_snp)]\n",
    "    # certify we want 50k data that doesn't overlap with cryptic\n",
    "    overlap = variants[~variants['sample_id'].isin(cryptic['ENA_RUN'])]\n",
    "    overlap = overlap['sample_id'].unique()\n",
    "\n",
    "    for x in tqdm(overlap):\n",
    "    # for x in tqdm(variants['sample_id'].unique()):\n",
    "        if x in variants['sample_id'].tolist():\n",
    "            aa.append(compare_snp_lists_with_values_optimized(all_snp, variants[variants['sample_id']==x]['SNP'].to_list(), variants[variants['sample_id']==x]['freq'].to_list()))\n",
    "            # print(variants[variants['sample_id']==x]['drugs'].unique())\n",
    "            result_ = transform_array(variants[variants['sample_id']==x]['drugs'].unique())\n",
    "            # print(result_)\n",
    "            # print(dr_list[0])\n",
    "            if dr_list[0] in result_:\n",
    "                dr.append(1)\n",
    "                # print('1') \n",
    "            else:\n",
    "                dr.append(0)\n",
    "                # print('0')\n",
    "        else:\n",
    "            # aa.append([0]*len(all_snp))\n",
    "            pass\n",
    "        # print('SNP')\n",
    "    aa_array = np.array(aa)\n",
    "    aa_array[aa_array < 0.8] = 0\n",
    "    aa_array[aa_array >= 0.8] = 1\n",
    "\n",
    "    return aa_array, dr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### RIF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0,)\n",
      "0\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "0it [00:00, ?it/s]\n"
     ]
    }
   ],
   "source": [
    "drug = 'RIF'\n",
    "dr_list = ['rifampicin']\n",
    "\n",
    "# df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "gene_list = ['rpoA', 'rpoB', 'rpoC']\n",
    "# df_emb = df[df[f'{drug}_MIC'].isin(['>4','4.0', '2.0', '1.0', '0.5', '0.25', '0.12', '<=0.06'])]\n",
    "df_emb = df[df[f'{drug}_MIC'].isin(['>4','4.0', '2.0', '1.0', '0.5', '0.25', '0.12', '0.06','<=0.06'])]\n",
    "# df_emb = df[df[f'{drug}_MIC'].isin(['>8','8.0','4.0', '2.0', '1.0', '0.5', '0.25', '0.12', '0.06','<=0.06'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = f'{drug}_MIC'\n",
    "    if row[x] == '>4' :\n",
    "        df_emb.loc[i, f'{x}'] = '8'\n",
    "    elif row[x] == '>8':\n",
    "        df_emb.loc[i, f'{x}'] = '16'\n",
    "    elif row[x] == '0.06':\n",
    "        df_emb.loc[i, f'{x}'] = '0.0625'\n",
    "        # df_emb.loc[i, f'{x}'] = '0.03'\n",
    "    elif row[x] == '0.12':\n",
    "        df_emb.loc[i, f'{x}'] = '0.125'\n",
    "    elif row[x] == '<=0.06':\n",
    "        df_emb.loc[i, f'{x}'] = '0.03125'\n",
    "        \n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "# aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))    \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ETH\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0,)\n",
      "0\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "0it [00:00, ?it/s]\n"
     ]
    }
   ],
   "source": [
    "drug = 'ETH'\n",
    "dr_list = ['ethionamide']\n",
    "gene_list = ['ethA', 'inhA']\n",
    "# df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "    \n",
    "# df_emb = df[df[f'{drug}_MIC'].isin(['>8','8.0', '4.0', '2.0', '1.0', '0.5', '0.25', '<=0.25'])]\n",
    "df_emb = df[df[f'{drug}_MIC'].isin(['>8','8.0', '4.0', '2.0', '1.0', '0.5'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = f'{drug}_MIC'\n",
    "    if row[x] == '>8' :\n",
    "        df_emb.loc[i, f'{x}'] = '16'\n",
    "    elif row[x] == '<=0.25':\n",
    "        df_emb.loc[i, f'{x}'] = '0.25'\n",
    "        \n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "# aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))    \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### INH\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0,)\n",
      "0\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "0it [00:00, ?it/s]\n"
     ]
    }
   ],
   "source": [
    "drug = 'INH'\n",
    "gene_list = ['inhA', 'katG','kasA']\n",
    "dr_list = ['isoniazid']\n",
    "# df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "# df_emb = df[df[f'{drug}_MIC'].isin(['>1.6','0.8', '0.4', '0.2', '0.1', '0.05', '<=0.025'])]\n",
    "df_emb = df[df[f'{drug}_MIC'].isin(['>1.6','1.6','0.8', '0.4', '0.2', '0.1', '0.05', '<=0.025'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = f'{drug}_MIC'\n",
    "    if row[x] == '>1.6' :\n",
    "        df_emb.loc[i, f'{x}'] = '3.2'\n",
    "    elif row[x] == '<=0.025':\n",
    "        # df_emb.loc[i, f'{x}'] = '0.0125'\n",
    "        df_emb.loc[i, f'{x}'] = '0.025'\n",
    "        \n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "# aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))    \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### RFB -- skipped\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0,)\n",
      "0\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "0it [00:00, ?it/s]\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'rifabutin'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/mnt/storageG1/lwang/miniconda3/envs/ml-g1/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3360\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3361\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3362\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/mnt/storageG1/lwang/miniconda3/envs/ml-g1/lib/python3.9/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m/mnt/storageG1/lwang/miniconda3/envs/ml-g1/lib/python3.9/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'rifabutin'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_13242/519754543.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     39\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     40\u001b[0m \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mf'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'wb'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 41\u001b[0;31m     \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msave\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdrs_cryptic\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34mf'{dr_list[0]}'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     42\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     43\u001b[0m \u001b[0;31m# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/mnt/storageG1/lwang/miniconda3/envs/ml-g1/lib/python3.9/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3456\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnlevels\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3457\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3458\u001b[0;31m             \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3459\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3460\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/mnt/storageG1/lwang/miniconda3/envs/ml-g1/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3361\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3362\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3363\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3364\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3365\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0misna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhasnans\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'rifabutin'"
     ]
    }
   ],
   "source": [
    "drug = 'RFB'\n",
    "res_thresh = 0.12\n",
    "dr_list = ['rifabutin']\n",
    "gene_list = ['rpoA', 'rpoB', 'rpoC']\n",
    "# df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "df_emb = df[df[f'{drug}_MIC'].isin(['>2','2.0', '1.0', '0.5', '0.25', '0.12'])]\n",
    "# df_emb = df[df[f'{drug}_MIC'].isin(['>2','2.0', '1.0', '0.5', '0.25', '0.12', '<=0.06'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = f'{drug}_MIC'\n",
    "    if row[x] == '>2' :\n",
    "        df_emb.loc[i, f'{x}'] = '4'\n",
    "    elif row[x] == '0.12':\n",
    "        df_emb.loc[i, f'{x}'] = '0.125'\n",
    "    elif row[x] == '<=0.06':\n",
    "        df_emb.loc[i, f'{x}'] = '0.03125'\n",
    "        \n",
    "        \n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "# aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "    \n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))    \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### MXF\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0,)\n",
      "0\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "0it [00:00, ?it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0,)\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "0it [00:00, ?it/s]\n"
     ]
    }
   ],
   "source": [
    "drug = 'MXF'\n",
    "dr_list = ['moxifloxacin']\n",
    "gene_list = ['gyrA', 'gyrB']\n",
    "res_thresh = 1\n",
    "df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "df_emb = df[df[f'{drug}_MIC'].isin(['>4','4.0', '2.0', '1.0', '0.5', '0.25', '0.12'])]\n",
    "# df_emb = df[df[f'{drug}_MIC'].isin(['>4','4', '2', '1', '0.5', '0.25', '0.12', '<=0.06'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = f'{drug}_MIC'\n",
    "    if row[x] == '>4' :\n",
    "        df_emb.loc[i, f'{x}'] = '8'\n",
    "    elif row[x] == '<=0.06':\n",
    "        df_emb.loc[i, f'{x}'] = '0.03'\n",
    "    elif row[x] == '0.12':\n",
    "        df_emb.loc[i, f'{x}'] = '0.125'\n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float') \n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))        \n",
    "    \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### KAN \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0,)\n",
      "0\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "0it [00:00, ?it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0,)\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "0it [00:00, ?it/s]\n"
     ]
    }
   ],
   "source": [
    "drug = 'KAN'\n",
    "dr_list = ['kanamycin']\n",
    "gene_list = ['rrs', 'eis']\n",
    "res_thresh = 4\n",
    "df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "df_emb = df[df[f'{drug}_MIC'].isin(['>16','16.0', '8.0', '4.0', '2.0', '<=1'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = f'{drug}_MIC'\n",
    "    if row[x] == '>16' :\n",
    "        df_emb.loc[i, f'{x}'] = '32'\n",
    "    elif row[x] == '<=1':\n",
    "        df_emb.loc[i, f'{x}'] = '1.0'\n",
    "        # df_emb.loc[i, f'{x}'] = '0.5'\n",
    "        \n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "    \n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))    \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### LEV\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "EOL while scanning string literal (2805265129.py, line 17)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"/tmp/ipykernel_13242/2805265129.py\"\u001b[0;36m, line \u001b[0;32m17\u001b[0m\n\u001b[0;31m    df_emb.loc[i, f'{x}'] = '0.125\u001b[0m\n\u001b[0m                                  ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m EOL while scanning string literal\n"
     ]
    }
   ],
   "source": [
    "drug = 'LEV'\n",
    "# gene_list = ['pgi', 'fadE4', 'php', 'cyp132', 'pckA', 'rpmB1', 'pfkB', 'acg', 'ctpF', 'cyp132', 'pckA', 'pfk']\n",
    "# gene_list = np.unique(variants['gene'])\n",
    "dr_list = ['levofloxacin']\n",
    "gene_list = ['gyrA', 'gyrB']\n",
    "res_thresh = 1\n",
    "df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "df_emb = df[df[f'{drug}_MIC'].isin(['>8','8.0', '4.0', '2.0', '1.0', '0.5', '0.25', '<=0.12'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = f'{drug}_MIC'\n",
    "    if row[x] == '>8' :\n",
    "        df_emb.loc[i, f'{x}'] = '16'\n",
    "    elif row[x] == '<=0.12':\n",
    "        df_emb.loc[i, f'{x}'] = '0.125\n",
    "        # df_emb.loc[i, f'{x}'] = '0.06'\n",
    "        \n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))    \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AMI\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "notebookRunGroups": {
     "groupValue": "2"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_90156/2595866171.py:20: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(306,)\n",
      "306\n",
      "11429\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 11429/11429 [00:35<00:00, 320.70it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(306,)\n",
      "306\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 124016/124016 [1:35:41<00:00, 21.60it/s]\n"
     ]
    }
   ],
   "source": [
    "# gene_list = ['pgi', 'fadE4', 'php', 'cyp132', 'pckA', 'rpmB1', 'pfkB', 'acg', 'ctpF', 'cyp132', 'pckA', 'pfk']\n",
    "# gene_list = np.unique(variants['gene'])\n",
    "dr_list = ['amikacin']\n",
    "drug = 'AMI'\n",
    "# gene_list = np.unique(variants['gene'])\n",
    "gene_list = ['rrs', 'eis']\n",
    "\n",
    "res_thresh = 1\n",
    "df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "df_emb = df[df[f'{drug}_MIC'].isin(['>8','8.0', '4.0', '2.0', '1.0', '0.5', '<=0.25'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = f'{drug}_MIC'\n",
    "    if row[x] == '>8' :\n",
    "        df_emb.loc[i, f'{x}'] = '16'\n",
    "    elif row[x] == '<=0.25':\n",
    "        df_emb.loc[i, f'{x}'] = '0.25'\n",
    "        # df_emb.loc[i, f'{x}'] = '0.125'\n",
    "        \n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "# aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))        \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### BDQ\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_90156/1471731688.py:21: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(234,)\n",
      "234\n",
      "10334\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 10334/10334 [00:01<00:00, 8051.41it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(234,)\n",
      "234\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 11552/11552 [00:53<00:00, 214.13it/s]\n"
     ]
    }
   ],
   "source": [
    "drug = 'BDQ'\n",
    "gene_list = [ 'Rv0678', 'atpE', 'pepQ']\n",
    "res_thresh = 0.25\n",
    "dr_list = ['bedaquiline']\n",
    "df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "df_emb = df[df[f'{drug}_MIC'].isin(['>1', '1.0', '0.5', '0.25', '0.12', '0.06', '0.03', '<=0.015'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = f'{drug}_MIC'\n",
    "    if row[x] == '>1' :\n",
    "        df_emb.loc[i, f'{x}'] = '2'\n",
    "    elif row[x] == '0.12':\n",
    "        df_emb.loc[i, f'{x}'] = '0.125'\n",
    "    elif row[x] == '0.06':\n",
    "        df_emb.loc[i, f'{x}'] = '0.0625'\n",
    "    elif row[x] == '0.03':\n",
    "        df_emb.loc[i, f'{x}'] = '0.03125'\n",
    "    elif row[x] == '<=0.015':\n",
    "        df_emb.loc[i, f'{x}'] = '0.015625'\n",
    "        \n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "# aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "    \n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))    \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CFZ\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "drug = 'CFZ'\n",
    "dr_list = ['clofazimine']\n",
    "# gene_list = np.unique(variants['gene'])\n",
    "gene_list = [ 'Rv0678', 'pepQ']\n",
    "res_thresh = 0.25\n",
    "df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "df_emb = df[df[f'{drug}_MIC'].isin(['>2','2.0', '1.0', '0.5', '0.25', '0.12', '0.06', '0.03', '<=0.015'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = f'{drug}_MIC'\n",
    "    if row[x] == '>1' :\n",
    "        df_emb.loc[i, f'{x}'] = '2'\n",
    "    elif row[x] == '0.12':\n",
    "        df_emb.loc[i, f'{x}'] = '0.125'\n",
    "    elif row[x] == '0.06':\n",
    "        df_emb.loc[i, f'{x}'] = '0.0625'\n",
    "    elif row[x] == '0.03':\n",
    "        df_emb.loc[i, f'{x}'] = '0.03125'\n",
    "    elif row[x] == '<=0.015':\n",
    "        df_emb.loc[i, f'{x}'] = '0.015625'\n",
    "        \n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "    \n",
    "    \n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))    \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### DLM\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_90156/2468873758.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(906,)\n",
      "906\n",
      "8807\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 8807/8807 [00:06<00:00, 1306.31it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(906,)\n",
      "906\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 49653/49653 [21:10<00:00, 39.10it/s]\n"
     ]
    }
   ],
   "source": [
    "drug = 'DLM'\n",
    "dr_list = ['delamanid']\n",
    "res_thresh = 0.12\n",
    "gene_list = ['fgd1','fbiA','fbiB','fbiC','fbiD','ddn']\n",
    "# gene_list = variants['gene'].unique()\n",
    "df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "df_emb = df[df[f'{drug}_MIC'].isin(['>0.5', '0.5', '0.25', '0.12', '0.06', '0.03', '0.015', '<=0.015'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = f'{drug}_MIC'\n",
    "    if row[x] == '>0.5' :\n",
    "        df_emb.loc[i, f'{x}'] = '1'\n",
    "    elif row[x] == '<=0.015':\n",
    "        df_emb.loc[i, f'{x}'] = '0.0078125'\n",
    "    elif row[x] == '0.12':\n",
    "        df_emb.loc[i, f'{x}'] = '0.125'        \n",
    "    elif row[x] == '0.06':\n",
    "        df_emb.loc[i, f'{x}'] = '0.0625'\n",
    "    elif row[x] == '0.03':\n",
    "        df_emb.loc[i, f'{x}'] = '0.03125'\n",
    "    elif row[x] == '0.015':\n",
    "        df_emb.loc[i, f'{x}'] = '0.015625'        \n",
    "        \n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "# aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "    \n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))    \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### EMB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_90156/1172994841.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1710,)\n",
      "1710\n",
      "11362\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 11362/11362 [01:07<00:00, 168.37it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1710,)\n",
      "1710\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 122744/122744 [3:02:29<00:00, 11.21it/s] \n"
     ]
    }
   ],
   "source": [
    "drug = 'EMB'\n",
    "dr_list = ['ethambutol']\n",
    "res_thresh = 8\n",
    "gene_list = ['embB', 'embA', 'embC']\n",
    "# gene_list = variants['gene'].unique()\n",
    "df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "df_emb = df[df['EMB_MIC'].isin(['>8','8.0', '4.0', '2.0', '1.0', '0.5'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = 'EMB_MIC'\n",
    "    if row[x] == '>8' :\n",
    "        df_emb.loc[i, f'{x}'] = '16'\n",
    "    elif row[x] == '<=0.25':\n",
    "        df_emb.loc[i, f'{x}'] = '0.125'\n",
    "df_emb[f'{drug}_MIC'] = df_emb[f'{drug}_MIC'].astype('float')\n",
    "\n",
    "cryptic = df_emb\n",
    "aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "aa_array_50k, drs_50k = data_prep_50k(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_cryptic)\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}-tbp.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_cryptic[f'{dr_list[0]}']))    \n",
    "    \n",
    "    \n",
    "# with open(f'./generated_data18122024/all_sample_snps_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_50k)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_50k_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sanity check with the EMB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "remove_from_50k = ['ERR1213824', 'ERR1213825', 'ERR1213826', 'ERR1213827', 'ERR1213828', 'ERR1213829', 'ERR1213830', 'ERR1213831', 'ERR1213832', 'ERR1213833', 'ERR1213834', 'ERR1213835', 'ERR1213836', 'ERR1213837', 'ERR1213838', 'ERR1213839', 'ERR1213840', 'ERR1213841', 'ERR1213842', 'ERR1213843', 'ERR1213844', 'ERR1213845', 'ERR1213846', 'ERR1213847', 'ERR1213848', 'ERR1213849', 'ERR1213850', 'ERR1213851', 'ERR1213852', 'ERR1213853', 'ERR1213854', 'ERR1213855', 'ERR1213856', 'ERR1213857', 'ERR1213858', 'ERR1213859', 'ERR1213860', 'ERR1213861', 'ERR1213862', 'ERR1213863', 'ERR1213864', 'ERR1213865', 'ERR1213866', 'ERR1213867', 'ERR1213868', 'ERR1213869', 'ERR1213870', 'ERR1213871', 'ERR1213872', 'ERR1213873', 'ERR1213874', 'ERR1213875', 'ERR1213876', 'ERR1213877', 'ERR1213878', 'ERR1213879', 'ERR1213880', 'ERR1213881', 'ERR1213882', 'ERR1213883', 'ERR1213884', 'ERR1213885', 'ERR1213886', 'ERR1213887', 'ERR1213888', 'ERR1213889', 'ERR1213890', 'ERR1213891', 'ERR1213892', 'ERR1213893', 'ERR1213894', 'ERR1213895', 'ERR1213896', 'ERR1213897', 'ERR1213898', 'ERR1213899', 'ERR1213900', 'ERR1213901', 'ERR1213902', 'ERR1213903', 'ERR1213904', 'ERR1213905', 'ERR1213906', 'ERR1213907', 'ERR1213908', 'ERR1213909', 'ERR1213910', 'ERR1213911', 'ERR1213912', 'ERR1213913', 'ERR1213914', 'ERR1213915', 'ERR1213916', 'ERR1213917', 'ERR1213918', 'ERR1213919', 'ERR1213920', 'ERR1213921', 'ERR1213922', 'ERR1213923', 'ERR1213924', 'ERR1213925', 'ERR1213926', 'ERR1213927', 'ERR1213928', 'ERR1213929', 'ERR1213930', 'ERR1213931', 'ERR1213932', 'ERR1213933', 'ERR1213934', 'ERR1213935', 'ERR1213936', 'ERR1213937', 'ERR1213938', 'ERR1213939', 'ERR1213940', 'ERR1213941', 'ERR1213942', 'ERR1213943', 'ERR1213944', 'ERR1213945', 'ERR1213946', 'ERR1213947', 'ERR1213948', 'ERR1213949', 'ERR1213950']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "variants = pd.read_csv('../variants_full.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>genome_pos</th>\n",
       "      <th>gene</th>\n",
       "      <th>change</th>\n",
       "      <th>freq</th>\n",
       "      <th>type</th>\n",
       "      <th>sublin</th>\n",
       "      <th>drtype</th>\n",
       "      <th>drugs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR9787047</td>\n",
       "      <td>6446</td>\n",
       "      <td>gyrB</td>\n",
       "      <td>p.Ala403Ser</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>lineage6.1.1</td>\n",
       "      <td>Sensitive</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR9787047</td>\n",
       "      <td>7362</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>p.Glu21Gln</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>lineage6.1.1</td>\n",
       "      <td>Sensitive</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR9787047</td>\n",
       "      <td>7585</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>p.Ser95Thr</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>lineage6.1.1</td>\n",
       "      <td>Sensitive</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR9787047</td>\n",
       "      <td>8493</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>p.Leu398Phe</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>lineage6.1.1</td>\n",
       "      <td>Sensitive</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR9787047</td>\n",
       "      <td>9143</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>c.1842T&gt;C</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>lineage6.1.1</td>\n",
       "      <td>Sensitive</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7420991</th>\n",
       "      <td>ERR718460</td>\n",
       "      <td>4328399</td>\n",
       "      <td>ethA</td>\n",
       "      <td>c.-926C&gt;T</td>\n",
       "      <td>0.986486</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>lineage2.2.1</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7420992</th>\n",
       "      <td>ERR718460</td>\n",
       "      <td>4338365</td>\n",
       "      <td>whiB6</td>\n",
       "      <td>p.Cys53Arg</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>lineage2.2.1</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7420993</th>\n",
       "      <td>ERR718460</td>\n",
       "      <td>4338595</td>\n",
       "      <td>whiB6</td>\n",
       "      <td>c.-75delG</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>lineage2.2.1</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7420994</th>\n",
       "      <td>ERR718460</td>\n",
       "      <td>4407588</td>\n",
       "      <td>gid</td>\n",
       "      <td>c.615A&gt;G</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>lineage2.2.1</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7420995</th>\n",
       "      <td>ERR718460</td>\n",
       "      <td>4407927</td>\n",
       "      <td>gid</td>\n",
       "      <td>p.Glu92Asp</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>lineage2.2.1</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7415787 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          sample_id  genome_pos   gene       change      freq  \\\n",
       "0        ERR9787047        6446   gyrB  p.Ala403Ser  1.000000   \n",
       "1        ERR9787047        7362   gyrA   p.Glu21Gln  1.000000   \n",
       "2        ERR9787047        7585   gyrA   p.Ser95Thr  1.000000   \n",
       "3        ERR9787047        8493   gyrA  p.Leu398Phe  1.000000   \n",
       "4        ERR9787047        9143   gyrA    c.1842T>C  1.000000   \n",
       "...             ...         ...    ...          ...       ...   \n",
       "7420991   ERR718460     4328399   ethA    c.-926C>T  0.986486   \n",
       "7420992   ERR718460     4338365  whiB6   p.Cys53Arg  1.000000   \n",
       "7420993   ERR718460     4338595  whiB6    c.-75delG  1.000000   \n",
       "7420994   ERR718460     4407588    gid     c.615A>G  1.000000   \n",
       "7420995   ERR718460     4407927    gid   p.Glu92Asp  1.000000   \n",
       "\n",
       "                          type        sublin     drtype drugs  \n",
       "0             missense_variant  lineage6.1.1  Sensitive   NaN  \n",
       "1             missense_variant  lineage6.1.1  Sensitive   NaN  \n",
       "2             missense_variant  lineage6.1.1  Sensitive   NaN  \n",
       "3             missense_variant  lineage6.1.1  Sensitive   NaN  \n",
       "4           synonymous_variant  lineage6.1.1  Sensitive   NaN  \n",
       "...                        ...           ...        ...   ...  \n",
       "7420991  upstream_gene_variant  lineage2.2.1      Other   NaN  \n",
       "7420992       missense_variant  lineage2.2.1      Other   NaN  \n",
       "7420993  upstream_gene_variant  lineage2.2.1      Other   NaN  \n",
       "7420994     synonymous_variant  lineage2.2.1      Other   NaN  \n",
       "7420995       missense_variant  lineage2.2.1      Other   NaN  \n",
       "\n",
       "[7415787 rows x 9 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants[~variants['sample_id'].isin(remove_from_50k)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 50k with samples removed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def data_prep_50k_mod(cryptic, gene_list, dr_list):\n",
    "    # overlap = set(variants['sample_id']).intersection(set(cryptic['ENA_RUN'].to_list()))\n",
    "    # variants = variants[variants['drugs'].isin(['ethambutol'])]\n",
    "    # variants = variants[variants['sample_id'].isin(overlap)]\n",
    "    # variants['SNP'] = variants['gene'] + '-'+ variants['change']\n",
    "    variants = pd.read_csv('../variants_full.csv')\n",
    "    variants = variants[variants['gene'] != 'PPE35']\n",
    "    variants = variants[variants['type'] != 'synonymous_variant']\n",
    "    variants = variants[variants['type'] != 'non_coding_transcript_exon_variant']\n",
    "\n",
    "    overlap = set(variants['sample_id']).intersection(set(cryptic['ENA_RUN'].to_list()))\n",
    "    # variants = variants[variants['drugs'].isin(['ethambutol'])]\n",
    "    variants = variants[variants['gene'].isin(gene_list)]\n",
    "    variants = variants[variants['sample_id'].isin(overlap)]\n",
    "    variants['SNP'] = variants['gene'] + '-'+ variants['change']\n",
    "    # crypticSNPnames = np.load('crypticSNPnames.npy', allow_pickle=True)\n",
    "    # variants = variants[variants['SNP'].isin(crypticSNPnames)]\n",
    "    print(variants['SNP'].unique().shape)\n",
    "    # print(variants.shape)\n",
    "    # print(variants['sample_id'].unique().shape)\n",
    "    def compare_snp_lists_with_values_optimized(set_list, query_list, values_list):\n",
    "        # Create a dictionary from query_list and values_list for direct mapping\n",
    "        query_dict = dict(zip(query_list, values_list))\n",
    "        # Use list comprehension to build the output list directly\n",
    "        output_list = [query_dict.get(snp, 0) for snp in set_list]\n",
    "        return output_list\n",
    "    # Example usage\n",
    "    # set_list = ['SNP1', 'SNP2', 'SNP3', 'SNP4']\n",
    "    # query_list = ['SNP2', 'SNP4']\n",
    "    # values_list = [5, 10]  # Corresponding values for 'SNP2' and 'SNP4'\n",
    "    # output_list = compare_snp_lists_with_values_optimized(set_list, query_list, values_list)\n",
    "    # print(output_list)  # Expected output: [0, 5, 0, 10]# Getting all snp data\n",
    "    aa = []\n",
    "    dr = []\n",
    "    \n",
    "    all_snp = variants['SNP'].unique() # here is a list of all snps values title for the row in the final table\n",
    "    np.save('all_snp_emb.npy', all_snp)\n",
    "    print(len(all_snp))\n",
    "### 50k data reload\n",
    "    variants = pd.read_csv('../variants_full.csv')\n",
    "    remove_from_50k = ['ERR1213824', 'ERR1213825', 'ERR1213826', 'ERR1213827', 'ERR1213828', 'ERR1213829', 'ERR1213830', 'ERR1213831', 'ERR1213832', 'ERR1213833', 'ERR1213834', 'ERR1213835', 'ERR1213836', 'ERR1213837', 'ERR1213838', 'ERR1213839', 'ERR1213840', 'ERR1213841', 'ERR1213842', 'ERR1213843', 'ERR1213844', 'ERR1213845', 'ERR1213846', 'ERR1213847', 'ERR1213848', 'ERR1213849', 'ERR1213850', 'ERR1213851', 'ERR1213852', 'ERR1213853', 'ERR1213854', 'ERR1213855', 'ERR1213856', 'ERR1213857', 'ERR1213858', 'ERR1213859', 'ERR1213860', 'ERR1213861', 'ERR1213862', 'ERR1213863', 'ERR1213864', 'ERR1213865', 'ERR1213866', 'ERR1213867', 'ERR1213868', 'ERR1213869', 'ERR1213870', 'ERR1213871', 'ERR1213872', 'ERR1213873', 'ERR1213874', 'ERR1213875', 'ERR1213876', 'ERR1213877', 'ERR1213878', 'ERR1213879', 'ERR1213880', 'ERR1213881', 'ERR1213882', 'ERR1213883', 'ERR1213884', 'ERR1213885', 'ERR1213886', 'ERR1213887', 'ERR1213888', 'ERR1213889', 'ERR1213890', 'ERR1213891', 'ERR1213892', 'ERR1213893', 'ERR1213894', 'ERR1213895', 'ERR1213896', 'ERR1213897', 'ERR1213898', 'ERR1213899', 'ERR1213900', 'ERR1213901', 'ERR1213902', 'ERR1213903', 'ERR1213904', 'ERR1213905', 'ERR1213906', 'ERR1213907', 'ERR1213908', 'ERR1213909', 'ERR1213910', 'ERR1213911', 'ERR1213912', 'ERR1213913', 'ERR1213914', 'ERR1213915', 'ERR1213916', 'ERR1213917', 'ERR1213918', 'ERR1213919', 'ERR1213920', 'ERR1213921', 'ERR1213922', 'ERR1213923', 'ERR1213924', 'ERR1213925', 'ERR1213926', 'ERR1213927', 'ERR1213928', 'ERR1213929', 'ERR1213930', 'ERR1213931', 'ERR1213932', 'ERR1213933', 'ERR1213934', 'ERR1213935', 'ERR1213936', 'ERR1213937', 'ERR1213938', 'ERR1213939', 'ERR1213940', 'ERR1213941', 'ERR1213942', 'ERR1213943', 'ERR1213944', 'ERR1213945', 'ERR1213946', 'ERR1213947', 'ERR1213948', 'ERR1213949', 'ERR1213950']\n",
    "    variants = variants[~variants['sample_id'].isin(remove_from_50k)]\n",
    "    variants = variants[variants['gene'] != 'PPE35']\n",
    "    variants = variants[variants['type'] != 'synonymous_variant']\n",
    "    variants = variants[variants['type'] != 'non_coding_transcript_exon_variant']\n",
    "    variants['SNP'] = variants['gene'] + '-'+ variants['change']\n",
    "    variants = variants[variants['SNP'].isin(all_snp)]\n",
    "    # certify we want 50k data that doesn't overlap with cryptic\n",
    "    overlap = variants[~variants['sample_id'].isin(cryptic['ENA_RUN'])]\n",
    "    overlap = overlap['sample_id'].unique()\n",
    "\n",
    "    for x in tqdm(overlap):\n",
    "    # for x in tqdm(variants['sample_id'].unique()):\n",
    "        if x in variants['sample_id'].tolist():\n",
    "            aa.append(compare_snp_lists_with_values_optimized(all_snp, variants[variants['sample_id']==x]['SNP'].to_list(), variants[variants['sample_id']==x]['freq'].to_list()))\n",
    "            result_ = transform_array(variants[variants['sample_id']==x]['drugs'].unique())\n",
    "            # print(result_)\n",
    "            # print(dr_list[0])\n",
    "            if dr_list[0] in result_:\n",
    "                dr.append(1)\n",
    "                # print('1') \n",
    "            else:\n",
    "                dr.append(0)\n",
    "        else:\n",
    "            # aa.append([0]*len(all_snp))\n",
    "            pass\n",
    "        # print('SNP')\n",
    "    aa_array = np.array(aa)\n",
    "    aa_array[aa_array < 0.8] = 0\n",
    "    aa_array[aa_array >= 0.8] = 1\n",
    "\n",
    "    return aa_array, dr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mnt/storageG1/lwang/miniconda3/envs/ml-g1/lib/python3.9/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n",
      "/mnt/storageG1/lwang/miniconda3/envs/ml-g1/lib/python3.9/site-packages/pandas/core/indexing.py:723: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value, self.name)\n",
      "/tmp/ipykernel_90156/3746604749.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_emb['EMB_MIC'] = df_emb['EMB_MIC'].astype('float')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1710,)\n",
      "1710\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 122619/122619 [3:01:54<00:00, 11.23it/s] \n"
     ]
    }
   ],
   "source": [
    "drug = 'EMB'\n",
    "dr_list = ['ethambutol']\n",
    "df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "gene_list = ['embB', 'embA', 'embC']\n",
    "df_emb = df[df['EMB_MIC'].isin(['>8','8.0', '4.0', '2.0', '1.0', '0.5'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = 'EMB_MIC'\n",
    "    if row[x] == '>8' :\n",
    "        df_emb.loc[i, f'{x}'] = '16'\n",
    "    elif row[x] == '<=0.25':\n",
    "        df_emb.loc[i, f'{x}'] = '0.125'\n",
    "        \n",
    "df_emb['EMB_MIC'] = df_emb['EMB_MIC'].astype('float') \n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "# aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "aa_array_50k, drs_50k = data_prep_50k_mod(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "# with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_cryptic)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "with open(f'./generated_data18122024/all_sample_snps_50k_{drug}_checks.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_50k)\n",
    "with open(f'./generated_data18122024/all_sample_drs_50k_{drug}_checks.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### removed samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def data_prep_50k_mod(cryptic, gene_list, dr_list):\n",
    "    # overlap = set(variants['sample_id']).intersection(set(cryptic['ENA_RUN'].to_list()))\n",
    "    # variants = variants[variants['drugs'].isin(['ethambutol'])]\n",
    "    # variants = variants[variants['sample_id'].isin(overlap)]\n",
    "    # variants['SNP'] = variants['gene'] + '-'+ variants['change']\n",
    "    variants = pd.read_csv('../variants_full.csv')\n",
    "    variants = variants[variants['gene'] != 'PPE35']\n",
    "    variants = variants[variants['type'] != 'synonymous_variant']\n",
    "    variants = variants[variants['type'] != 'non_coding_transcript_exon_variant']\n",
    "    overlap = set(variants['sample_id']).intersection(set(cryptic['ENA_RUN'].to_list()))\n",
    "    # variants = variants[variants['drugs'].isin(['ethambutol'])]\n",
    "    variants = variants[variants['gene'].isin(gene_list)]\n",
    "    variants = variants[variants['sample_id'].isin(overlap)]\n",
    "    variants['SNP'] = variants['gene'] + '-'+ variants['change']\n",
    "    # crypticSNPnames = np.load('crypticSNPnames.npy', allow_pickle=True)\n",
    "    # variants = variants[variants['SNP'].isin(crypticSNPnames)]\n",
    "    print(variants['SNP'].unique().shape)\n",
    "    # print(variants.shape)\n",
    "    # print(variants['sample_id'].unique().shape)\n",
    "    def compare_snp_lists_with_values_optimized(set_list, query_list, values_list):\n",
    "        # Create a dictionary from query_list and values_list for direct mapping\n",
    "        query_dict = dict(zip(query_list, values_list))\n",
    "        # Use list comprehension to build the output list directly\n",
    "        output_list = [query_dict.get(snp, 0) for snp in set_list]\n",
    "        return output_list\n",
    "    # Example usage\n",
    "    # set_list = ['SNP1', 'SNP2', 'SNP3', 'SNP4']\n",
    "    # query_list = ['SNP2', 'SNP4']\n",
    "    # values_list = [5, 10]  # Corresponding values for 'SNP2' and 'SNP4'\n",
    "    # output_list = compare_snp_lists_with_values_optimized(set_list, query_list, values_list)\n",
    "    # print(output_list)  # Expected output: [0, 5, 0, 10]# Getting all snp data\n",
    "    aa = []\n",
    "    dr = []\n",
    "    \n",
    "    all_snp = variants['SNP'].unique() # here is a list of all snps values title for the row in the final table\n",
    "    # np.save('all_snp_emb.npy', all_snp)\n",
    "    print(len(all_snp))\n",
    "### 50k data reload\n",
    "    variants = pd.read_csv('../variants_full.csv')\n",
    "    remove_from_50k = ['ERR1213824', 'ERR1213825', 'ERR1213826', 'ERR1213827', 'ERR1213828', 'ERR1213829', 'ERR1213830', 'ERR1213831', 'ERR1213832', 'ERR1213833', 'ERR1213834', 'ERR1213835', 'ERR1213836', 'ERR1213837', 'ERR1213838', 'ERR1213839', 'ERR1213840', 'ERR1213841', 'ERR1213842', 'ERR1213843', 'ERR1213844', 'ERR1213845', 'ERR1213846', 'ERR1213847', 'ERR1213848', 'ERR1213849', 'ERR1213850', 'ERR1213851', 'ERR1213852', 'ERR1213853', 'ERR1213854', 'ERR1213855', 'ERR1213856', 'ERR1213857', 'ERR1213858', 'ERR1213859', 'ERR1213860', 'ERR1213861', 'ERR1213862', 'ERR1213863', 'ERR1213864', 'ERR1213865', 'ERR1213866', 'ERR1213867', 'ERR1213868', 'ERR1213869', 'ERR1213870', 'ERR1213871', 'ERR1213872', 'ERR1213873', 'ERR1213874', 'ERR1213875', 'ERR1213876', 'ERR1213877', 'ERR1213878', 'ERR1213879', 'ERR1213880', 'ERR1213881', 'ERR1213882', 'ERR1213883', 'ERR1213884', 'ERR1213885', 'ERR1213886', 'ERR1213887', 'ERR1213888', 'ERR1213889', 'ERR1213890', 'ERR1213891', 'ERR1213892', 'ERR1213893', 'ERR1213894', 'ERR1213895', 'ERR1213896', 'ERR1213897', 'ERR1213898', 'ERR1213899', 'ERR1213900', 'ERR1213901', 'ERR1213902', 'ERR1213903', 'ERR1213904', 'ERR1213905', 'ERR1213906', 'ERR1213907', 'ERR1213908', 'ERR1213909', 'ERR1213910', 'ERR1213911', 'ERR1213912', 'ERR1213913', 'ERR1213914', 'ERR1213915', 'ERR1213916', 'ERR1213917', 'ERR1213918', 'ERR1213919', 'ERR1213920', 'ERR1213921', 'ERR1213922', 'ERR1213923', 'ERR1213924', 'ERR1213925', 'ERR1213926', 'ERR1213927', 'ERR1213928', 'ERR1213929', 'ERR1213930', 'ERR1213931', 'ERR1213932', 'ERR1213933', 'ERR1213934', 'ERR1213935', 'ERR1213936', 'ERR1213937', 'ERR1213938', 'ERR1213939', 'ERR1213940', 'ERR1213941', 'ERR1213942', 'ERR1213943', 'ERR1213944', 'ERR1213945', 'ERR1213946', 'ERR1213947', 'ERR1213948', 'ERR1213949', 'ERR1213950']\n",
    "    variants = variants[variants['sample_id'].isin(remove_from_50k)]\n",
    "    variants = variants[variants['gene'] != 'PPE35']\n",
    "    variants = variants[variants['type'] != 'synonymous_variant']\n",
    "    variants = variants[variants['type'] != 'non_coding_transcript_exon_variant']\n",
    "    variants['SNP'] = variants['gene'] + '-'+ variants['change']\n",
    "    variants = variants[variants['SNP'].isin(all_snp)]\n",
    "    # certify we want 50k data that doesn't overlap with cryptic\n",
    "    overlap = variants[~variants['sample_id'].isin(cryptic['ENA_RUN'])]\n",
    "    overlap = overlap['sample_id'].unique()\n",
    "    print(\"len(overlap)\",len(overlap))\n",
    "\n",
    "    for x in tqdm(overlap):\n",
    "    # for x in tqdm(variants['sample_id'].unique()):\n",
    "        if x in variants['sample_id'].tolist():\n",
    "            aa.append(compare_snp_lists_with_values_optimized(all_snp, variants[variants['sample_id']==x]['SNP'].to_list(), variants[variants['sample_id']==x]['freq'].to_list()))\n",
    "            \n",
    "            result_ = transform_array(variants[variants['sample_id']==x]['drugs'].unique())\n",
    "            # print(result_)\n",
    "            # print(dr_list[0])\n",
    "            if dr_list[0] in result_:\n",
    "                dr.append(1)\n",
    "                # print('1') \n",
    "            else:\n",
    "                dr.append(0)\n",
    "        else:\n",
    "            # aa.append([0]*len(all_snp))\n",
    "            pass\n",
    "        # print('SNP')\n",
    "    aa_array = np.array(aa)\n",
    "    aa_array[aa_array < 0.8] = 0\n",
    "    aa_array[aa_array >= 0.8] = 1\n",
    "\n",
    "    return aa_array, dr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mnt/storageG1/lwang/miniconda3/envs/ml-g1/lib/python3.9/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n",
      "/mnt/storageG1/lwang/miniconda3/envs/ml-g1/lib/python3.9/site-packages/pandas/core/indexing.py:723: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  iloc._setitem_with_indexer(indexer, value, self.name)\n",
      "/tmp/ipykernel_52736/1811954684.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_emb['EMB_MIC'] = df_emb['EMB_MIC'].astype('float')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1710,)\n",
      "1710\n",
      "len(overlap) 125\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 125/125 [00:00<00:00, 760.72it/s]\n"
     ]
    }
   ],
   "source": [
    "drug = 'EMB'\n",
    "dr_list = ['ethambutol']\n",
    "df = pd.read_csv('../CRyPTIC_reuse_table_20231208.csv')\n",
    "df = pd.read_csv('/mnt/storageG1/lwang/Projects/tb_dr_MIC3/CRyPTIC_reuse_table_20231208_cleaned+tbprofiler.csv')\n",
    "\n",
    "gene_list = ['embB', 'embA', 'embC']\n",
    "df_emb = df[df['EMB_MIC'].isin(['>8','8.0', '4.0', '2.0', '1.0', '0.5'])]\n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "for i, row in df_emb.iterrows():\n",
    "    x = 'EMB_MIC'\n",
    "    if row[x] == '>8' :\n",
    "        df_emb.loc[i, f'{x}'] = '16'\n",
    "    elif row[x] == '<=0.25':\n",
    "        df_emb.loc[i, f'{x}'] = '0.125'\n",
    "        \n",
    "df_emb['EMB_MIC'] = df_emb['EMB_MIC'].astype('float') \n",
    "# df_emb = df_emb[~df_emb['ENA_RUN'].isin(to_be_dropped)]\n",
    "\n",
    "# variants = pd.read_csv('variants_full.csv')\n",
    "# variants = variants[variants['type'] != 'synonymous_variant']\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['LOW','MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[~df_emb['EMB_PHENOTYPE_QUALITY'].isin(['MEDIUM'])]  # remove low and med quality\n",
    "# df_emb = df_emb[df_emb['ENA_RUN'].isin(samples)]\n",
    "cryptic = df_emb\n",
    "# aa_array_cryptic, drs_cryptic = data_prep_cryptic(cryptic, gene_list,dr_list)\n",
    "aa_array_50k, drs_50k = data_prep_50k_mod(cryptic, gene_list,dr_list)\n",
    "\n",
    "# encoded_mic = np.array([0 if value < 4 else 1 for value in mic_aa['EMB_MIC'].to_list()])\n",
    "\n",
    "# with open(f'./generated_data18122024/all_sample_snps_cryptic_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, aa_array_cryptic)\n",
    "# with open(f'./generated_data18122024/all_sample_drs_cryptic_{drug}.npy', 'wb') as f:\n",
    "#     np.save(f, np.array(drs_cryptic[f'{drug}_MIC']))\n",
    "with open(f'./generated_data18122024/all_sample_snps_50k_{drug}_checks-removed.npy', 'wb') as f:\n",
    "    np.save(f, aa_array_50k)\n",
    "with open(f'./generated_data18122024/all_sample_drs_50k_{drug}_checks-removed.npy', 'wb') as f:\n",
    "    np.save(f, np.array(drs_50k))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_values = ['8', '4', '≤1', '≤1', '2', '≤1', '≤1', '2', '2', '≤1', '≤1', '4', '2', '4', '≤1', '8', '2', '2', '4', '4', '8', '≤1', '≤1', '≤1', '2', '>','8', '4', '2', '4', '2', '≤1', '2', '2', '≤1', '4', '>','8', '2', '≤1', '4', '4', '8', '4', '2', '≤1', '≤1', '≤1', '≤1', '4', '4', '≤1', '>','8', '≤1', '≤1', '2', '2', '≤1', '≤1', '2', '2', '4', '≤1', '≤1', '2', '4', '≤1', '8', '2', '2', '4', '2', '≤1', '≤1', '≤1', '≤1', '≤1', '≤1', '≤1', '2', '4', '≤1', '4', '≤1', '≤1', '≤1', '≤1', '4', '2', '≤1', '≤1', '≤1', '2', '≤1', '≤1', '≤1', '≤1', '≤1', '2', '≤1', '≤1', '8', '2', '≤1', '8', '2', '4', '2', '≤1', '≤1', '2', '≤1', '≤1', '≤1', '≤1', '≤1', '≤1', '≤1', '≤1', '2', '≤1', '≤1', '≤1', '≤1', '≤1', '>','8', '8', '4', '4']\n",
    "for i, x in enumerate(test_values):\n",
    "    if x == '≤1':\n",
    "        test_values[i] = '1'"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ml-g1",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
